src/gui/src/main.cpp

```diff
@@ -63,14 +62,18 @@ void copy_qsettings(const QSettings &src, QSettings &dst)
 
 int main(int argc, char* argv[])
 {
-#ifdef WINAPI_XWINDOWS
-    // QApplication's constructor will call a fscking abort() if
-    // DISPLAY is bad. Let's check it first and handle it gracefully
-    if (!display_is_valid()) {
-        fprintf(stderr, "The InputLeap GUI requires the DISPLAY environment variable to be set. Quitting...\n");
-        return 1;
+#if (defined(WINAPI_XWINDOWS) || (defined(HAVE_LIBPORTAL_INPUTCAPTURE) || defined(HAVE_LIBPORTAL_SESSION_CONNECT_TO_EIS)))
+    const auto platformType = QGuiApplication::platformName();
+
+    if (platformType == "xcb") {
+        // We're running on X11, all good.
+        // Continue running.
+    } else if (platformType == "wayland") {
+        QMessageBox::information(nullptr, "Input Leap",
+                                 "You are using Wayland. Input Leap supports Wayland via `libei`, but not all desktop environment/window managers support our implementation at this time. Therefore, your mileage may vary.");
```

Comment:
Too long line, please wrap at 100 chars.

Reply:
@p12tic As someone who is still learning C++, would I need to use string concatenation, or not?

Reply:
@p12tic Found a way. Does this latest push resolve this conversation?